pymc3/distributions/timeseries.py

```diff
@@ -176,6 +184,75 @@ def logp(self, value):
         return at.sum(innov_like) + at.sum(init_like)
 
 
+class ARMARV(RandomVariable):
+    name = "ARMA"
+    ndim_supp = 0
+    ndims_params = [1, 1, 0, 0]
+    dtype = "floatX"
+    _print_name = ("ARMA", "\\operatorname{ARMA}")
+
+    def __call__(self, phi=[], theta=[], loc=[0.0], scale=[1.0], size=100, **kwargs) -> TensorVariable:
+        return super().__call__(phi, theta, loc, scale, size=size, **kwargs)
+
+    @classmethod
+    def rng_fn(
+            cls,
+            rng: np.random.default_rng(),
+            phi: np.ndarray,
+            theta: np.ndarray,
+            loc: np.ndarray,
+            scale: np.ndarray,
+            size: Tuple[int, ...],
+    ) -> np.ndarray:
+        phi = np.r_[1, -phi]
+        theta = np.r_[1, theta]
+        x = rng.normal(loc=loc, scale=scale, size=size)
+        return signal.lfilter(theta, phi, x, axis=0)
+
+
+ARMArv = ARMARV()
+
+class ARMA(Continuous):
+    rv_op = ARMArv
+
+    @classmethod
+    def dist(cls, P=None, Q=None, phi=None, theta=None, loc=0.0, scale=1.0,size=100, **kwargs):
```

Comment:
I have the same question, should we just use one set of args and have a docstring or do we want to support both?

For example in the PyMC3 beta distribution we support both `alpha` and `beta` params as well as `mu,` `sigma` but not sure if thats a good idea or not?

Not saying we need a change, just asking the same question

Reply:
so there's two sides to this, there's a reason i'm supporting both ways of parametrizing. say a user wants to generate ARMA(2,2) without knowing phi or theta, then P and Q are useful. But if they know phi and theta, and want to actually generate a specific ARMA (2,2) where say phi = [0.25,0.56] and theta = [-0.76,0.87] so both the ways might be useful, and we should leave it upto the user to decide what they want? and phi and theta are commonly used in the literature but, to avoid any risks whatsoever, I'll name the parameters as follows:

P - number(order) of autogressive lags
Q - number(order) of moving-average lags

phi (renamed as AR) - tensor of autoregressive coefficients
theta(renamed as MA) - tensor of moving-average coefficients

I think it would make sense to do this in AR(p) in timeseries.py as well?

ofcourse i might be wrong, let me know if this sounds reasonable?

Reply:
Sounds reasonable to support both parametrizations, but I would still consider to name them `lags_autoregressive`, `lags_movingaverage` (`coeff_autoregressive`, `coeff_movingaverage`) or similar. Otherwise consider lowercase letters - I think some linters might complain about uppercase letters.

Also a `, *,` to require kwargs might be a good idea.
One might argue, that it's a little verbose for the model definition, but the time series distribution is probably quite important for the users model, so it should be okay to be more verbose?